src/tests/token/erc20/test_erc20.cairo

```diff
@@ -96,10 +94,13 @@ fn test_allowance() {
 #[test]
 fn test_approve() {
     let mut state = setup();
-    testing::set_caller_address(OWNER());
+    let contract_address = test_address();
+    let mut spy = utils::spy_on(contract_address);
```

Comment:
This address is not the target address for component tests, but the address of the current context in foundry (starknet foundry contract running the tests). contract address is probably still not the best name, but I vote we keep it over target_address, since to me is a bit more confusing.

Reply:
Right, target is not a good name. `contract_address` works!